server/src/main/java/org/apache/druid/server/coordinator/CoordinatorDynamicConfig.java

```diff
@@ -315,7 +318,9 @@ public int getBalancerComputeThreads()
   @JsonProperty("killAllDataSources")
   public boolean isKillUnusedSegmentsInAllDataSources()
   {
-    return killUnusedSegmentsInAllDataSources;
+    return killUnusedSegmentsInAllDataSources != null
+           ? killUnusedSegmentsInAllDataSources
+           : specificDataSourcesToKillUnusedSegmentsIn.isEmpty();
```

Comment:
Can this be simplified as?

```suggestion
    return specificDataSourcesToKillUnusedSegmentsIn.isEmpty();
```

The method returns true when
- either `killUnusedSegmentsInAllDataSources = true` which implies that `specifiedDataSourcesToKillUnusedSegmentsIn` is empty, otherwise we would have thrown an exception
- or `killUnusedSegmentsInAllDataSources = null` and `specifiedDataSourcesToKillUnusedSegmentsIn` is empty

With these new defaults, maybe we don't even need `killAll` anymore?
If `killList` is empty, we kill all
Otherwise, we kill only the ones specified.
Unless I am missing a case.

Reply:
Great suggestion. This should simplify the user experience. I will do this